### PR TITLE
lib/server: tfgrid add updates

### PR DESCRIPTION
**Pull request by MarioBassem**
## Related issues:

- #27

### CODE REVIEW

**Comment by MarioBassem** [comment]
this pr should only be merged after using a version of the grid-client that has this change: https://github.com/threefoldtech/tfgrid-sdk-go/pull/109

**Comment by MarioBassem** [comment]
k8s worker or machine removals need this to be solved first (https://github.com/threefoldtech/tfgrid-sdk-go/issues/125)

File: server/clients/tfgrid/machines.go [hunk 1]
```diff
@@ -114,124 +126,423 @@ type Groups []Group
...
+		return MachinesModel{}, errors.Wrapf(err, "failed to add machine %s", params.Machine.Name)
 	}
 
-	// construct result
-	resModel, err := r.constructMachinesModelFromContracts(ctx, nodeDeploymentID, model.Name, net)
+	return c.MachinesGet(ctx, params.ModelName)
+}
+
+func (c *Client) MachineRemove(ctx context.Context, params RemoveMachineParams) (MachinesModel, error) {
+	log.Info().Msgf("removeing machine %s", params.MachineName)
```
Comment (brandonpille) [inline server/clients/tfgrid/machines.go:226]: Makes more sense to put these to Debug(). Can you also do it for other calls?

File: server/pkg/tfgrid/client.go [hunk 2]
```diff
@@ -44,13 +43,16 @@ func NewClient() *Client {
 	}
 }
 
-func generateProjectName(modelName string) (projectName string) {
-	return fmt.Sprintf("%s.web3proxy", modelName)
-}
+// func generateProjectName(modelName string) (projectName string) {
```
Comment (brandonpille) [inline server/pkg/tfgrid/client.go:46]: If we don't need it anymore remove please.

File: server/pkg/tfgrid/client.go [hunk 3]
```diff
@@ -172,9 +192,7 @@ func (c *Client) GatewayNameDeploy(ctx context.Context, model tfgridBase.Gateway
 		return tfgridBase.GatewayNameModel{}, pkg.ErrClientNotConnected{}
 	}
 
-	projectName := generateProjectName(model.Name)
-
-	return state.cl.GatewayNameDeploy(ctx, model, projectName)
+	return state.cl.GatewayNameDeploy(ctx, model)
```
Comment (brandonpille) [inline server/pkg/tfgrid/client.go:195]: Can you update the functions to use the new way of getting the client from state. See for example in tfchain:

```
// State from a connection. If no state is present, it is initialized
func State(conState jsonrpc.State) *TfchainState {
	raw, exists := conState[TfchainID]
	if !exists {
		ns := &TfchainState{
			client:  nil,
			network: tfchainTestnet,
		}
		conState[TfchainID] = ns
		return ns
	}
	ns, ok := raw.(*TfchainState)
	if !ok {
		// This means the invariant is violated, so panic here is ok
		panic("Invalid saved state for tfchain")
	}
	return ns
}

// Transer an amount of TFT from the loaded account to the destination.
func (c *Client) Transfer(ctx context.Context, conState jsonrpc.State, args Transfer) error {
	state := State(conState)
	if state.client == nil {
		return pkg.ErrClientNotConnected{}
	}

```

**Review by brandonpille — COMMENTED** [review]
Nice work. Some comments:

**Pull request merged**